### PR TITLE
feat: ai-prompt-template plugin

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Implement ai-prompt-template plugin.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/ai-prompt-template.lua [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+local function find_template(conf, template_name)
+    for _, template in ipairs(conf.templates) do
+        if template.name == template_name then
+            return template.template
+        end
+    end
+    return nil
+end
```
Comment (bzp2010) [inline apisix/plugins/ai-prompt-template.lua:108]: When there are a lot of templates, it's not a good idea to go through them completely every time. Do you think the plugin can be configured with more templates? Like dozens of them? 🤔

File: apisix/plugins/ai-prompt-template.lua [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if not template_name then
+        return 400, { message = "template name is missing in request." }
+    end
+
+    local template = find_template(conf, template_name)
+    if not template then
+        return 400, { message = "template: " .. template_name .. " not configured." }
+    end
+
+    local template_json = core.json.encode(template)
```
Comment (bzp2010) [inline apisix/plugins/ai-prompt-template.lua:125]: The json encode is executed every time, which may not be good for performance. I think they are cacheable, again do you think it is necessary to add caching now?

Comment (bzp2010) [reply apisix/plugins/ai-prompt-template.lua:125]: If you don't think this is a big issue right now, we can do it later.

**Review by bzp2010 — COMMENTED** [review]
LGTM

File: apisix/plugins/ai-prompt-template.lua (hunk 1 above)
Comment (shreemaan-abhishek) [reply apisix/plugins/ai-prompt-template.lua:115]: >  Like dozens of them?

possible, but there would be no more than 20 templates in normal usage scenarios from my speculation.

File: apisix/plugins/ai-prompt-template.lua (hunk 2 above)
Comment (shreemaan-abhishek) [reply apisix/plugins/ai-prompt-template.lua]: > If you don't think this is a big issue right now, we can do it later.

it's a simple thing so I did it right away.

File: apisix/plugins/ai-prompt-template.lua [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    schema   = schema,
+}
+
+local templates_lrucache = core.lrucache.new({
+    ttl = 300, count = 256
+})
+
+local templates_json_lrucache = core.lrucache.new({
+    ttl = 300, count = 256
+})
```
Comment (bzp2010) [inline apisix/plugins/ai-prompt-template.lua:86]: Does this cache work properly when configuring multiple instances of the plugin on multiple routes? The template name doesn't have a prefix, do they conflict when configuring multiple templates with the same name?

Comment (shreemaan-abhishek) [reply apisix/plugins/ai-prompt-template.lua:86]: it works, https://github.com/apache/apisix/pull/11517/commits/0c52a21ce6e8398d5181012cf6f196acfb55b5e2

**Review by bzp2010 — DISMISSED** [review]
Code LGTM

File: docs/en/latest/plugins/ai-prompt-template.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+using a template and allowing users to pass only the values for template variables.
+
+## Plugin Attributes
+
+| **Field**                             | **Type** | **Description**                                     | **Required** |
+| ------------------------------------- | -------- | --------------------------------------------------- | ------------ |
+| `templates`                           | Array    | An array of template objects                        | Yes          |
+| `templates.name`                      | String   | Name of the template.                               | Yes          |
+| `templates.template.model`            | String   | Model of the AI Model. Example: gpt-4, gpt-3.5      | Yes          |
+| `templates.template.messages.role`    | String   | Role of the message (`system`, `user`, `assistant`) | Yes          |
```
Comment (kayx23) [inline docs/en/latest/plugins/ai-prompt-template.md:42]:
```suggestion
| `templates.template.messages.role`    | String   | Role of the message, such as `system`, `user`, or `assistant`. | Yes          |
```

File: docs/en/latest/plugins/ai-prompt-template.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      "scheme": "https",
+      "pass_host": "node"
+    },
+    "plugins": {
+      "ai-prompt-template": {
+        "templates": [
+          {
+            "name": "level of detail",
+            "template": {
+              "model": "gpt-4",
```
Comment (kayx23) [inline docs/en/latest/plugins/ai-prompt-template.md:68]:
```suggestion
              "model": "gpt-4o",
```

File: docs/en/latest/plugins/ai-prompt-template.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    }
+  }'
+```
+
+Now send a request:
+
+```shell
+curl http://127.0.0.1:9080/v1/chat/completions -i -XPOST  -H 'Content-Type: application/json' -d '{
+  "template_name": "level of detail,
```
Comment (kayx23) [inline docs/en/latest/plugins/ai-prompt-template.md:87]:
```suggestion
  "template_name": "level of detail",
```

**Review by moonming — DISMISSED** [review]

**Review by kayx23 — APPROVED** [review]
doc LGTM

**Review by moonming — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**